### PR TITLE
Add camp.nook.nookdesktop

### PR DESCRIPTION
**Pull request by OpenSauce04**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. ~If not, I contacted upstream developers about submitting their software to Flathub. **Link:**~
- [ ] ~I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).~ I do not own the domain, however the original author of the project does.
- [ ] ~Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*~ The Nook Desktop project appears to be completely unmaintained, as I had originally submitted a PR to include a Linux build 3 months ago, and there was never a response. The original author has been MIA since the 11th of February, with no commits to any repos on GitHub.

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: camp.nook.nookdesktop.yml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+command: run.sh
+separate-locales: false
+
+finish-args:
+  # X11 + XShm access
+  - --share=ipc
+  - --socket=fallback-x11
+  # Wayland access
+  - --socket=wayland
```
Comment (hfiguiere) [inline camp.nook.nookdesktop.yml:19]: are you sure this work with wayland? Usually Electron doesn't

File: nook-desktop.desktop [hunk 2]
```diff
@@ -0,0 +1,6 @@
+[Desktop Entry]
+Name=Nook
+Exec=
```
Comment (hfiguiere) [inline nook-desktop.desktop:3]: exec is empty

File: camp.nook.nookdesktop.appdata.xml
Comment (hfiguiere) [inline camp.nook.nookdesktop.appdata.xml:1]: this file is never installed. That's the build failure.

File: camp.nook.nookdesktop.yml [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+      # Export desktop file
+      - install -Dm644 nook-desktop.desktop /app/share/applications/camp.nook.nookdesktop.desktop
+      # Install app wrapper
+      - install -Dm755 -t /app/bin/ ../run.sh
+    subdir: main
+
+    sources:
+
+      - type: git
+        url: https://github.com/OpenSauce04/nook-desktop.git
```
Comment (hfiguiere) [inline camp.nook.nookdesktop.yml:64]: why isn't this pulled from actual upstream?

File: camp.nook.nookdesktop.yml [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+      - generated-sources.json
+
+      # Wrapper to launch the app
+      - type: script
+        dest-filename: run.sh
+        commands:
+          - zypak-wrapper.sh /app/main/nook-desktop "$@"
+
+      - type: file
+      path: camp.nook.nookdesktop.appdata.xml
```
Comment (hfiguiere) [inline camp.nook.nookdesktop.yml:81]: the indent is incorrect. also that doesn't install it either. it is not sufficient.

File: nook-desktop.desktop (hunk 2 above)
Comment (OpenSauce04) [reply nook-desktop.desktop:3]: Actually looking back I'm pretty sure I misread it, I will fill it in

File: camp.nook.nookdesktop.yml [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.node18
+
+command: run.sh
+separate-locales: false
+
+finish-args:
+  # X11 + XShm access
+  - --share=ipc
+  - --socket=fallback-x11
```
Comment (hfiguiere) [inline camp.nook.nookdesktop.yml:17]:
```suggestion
  - --socket=x11
```

File: nook-desktop.desktop [hunk 6]
```diff
@@ -0,0 +1,6 @@
+[Desktop Entry]
+Name=Nook
+Exec=camp.nook.nookdesktop
```
Comment (hfiguiere) [inline nook-desktop.desktop:3]: there is no such executable. it should be what you put in `command` in the manifest.

**Comment by OpenSauce04** [comment]
bot, build camp.nook.nookdesktop

**Comment by OpenSauce04** [comment]
bot, build camp.nook.nookdesktop

**Comment by OpenSauce04** [comment]
Not sure why it can't get the image file, it definitely exists

**Comment by hfiguiere** [comment]
> Not sure why it can't get the image file, it definitely exists

```
$ wget https://i.imgur.com/NnNdJ0d.png
--2023-05-25 11:40:08--  https://i.imgur.com/NnNdJ0d.png
Resolving i.imgur.com (i.imgur.com)... 151.101.136.193
Connecting to i.imgur.com (i.imgur.com)|151.101.136.193|:443... connected.
HTTP request sent, awaiting response... 429 Unknown Error
2023-05-25 11:40:08 ERROR 429: Unknown Error.
```

**Comment by OpenSauce04** [comment]
bot, build camp.nook.nookdesktop

File: camp.nook.nookdesktop.yml (hunk 3 above)
Comment (barthalion) [reply camp.nook.nookdesktop.yml:64]: It's not a blocker but yes, it would be preferred to ship patches.

**Pull request closed without merging**